README.md

```diff
@@ -1 +1,415 @@
+```c
+void gpios_config(const struct device *gpio1, const struct device *gpio2){
+	int ret;
+	ret = gpio_pin_configure(gpio1, 0, GPIO_OUTPUT_ACTIVE);//green
+	if (ret != 0) {
+		return;
+	}
+		ret = gpio_pin_configure(gpio2, 10, GPIO_OUTPUT_ACTIVE);//green
+	if (ret != 0) {
+		return;
+	}
+	if(!device_is_ready(gpio1)){
+		return;
+	}
+	if(!device_is_ready(gpio2)){
+		return;
+	}
+
+
+}
+```
+This function configs gpios outputs for led controling.
+```c
+void adc_config(){
+	int err;
+	for (size_t i = 0U; i < ARRAY_SIZE(adc_channels); i++) {
+		if (!adc_is_ready_dt(&adc_channels[i])) {
+			LOG_INF("ADC controller device %s not ready\n", adc_channels[i].dev->name);
+			return 0;
+		}
+
+		err = adc_channel_setup_dt(&adc_channels[i]);
+		if (err < 0) {
+			LOG_INF("Could not setup channel #%d (%d)\n", i, err);
+			return 0;
+		}
+	}
+}
+```
+This function is used to configs analog to digital converter.
+
+```c
+int adc_val(){
+	uint16_t buf;
+	struct adc_sequence sequence = {
+		.buffer = &buf,
+		.buffer_size = sizeof(buf),
+	};
+	int err;
+	int moisture;
+	for (size_t i = 0U; i < ARRAY_SIZE(adc_channels); i++) {
+		int32_t val_mv;
+		(void)adc_sequence_init_dt(&adc_channels[i], &sequence);
+		err = adc_read_dt(&adc_channels[i], &sequence);
+		if (err < 0) {
+			
+			continue;
+		}
+		if (adc_channels[i].channel_cfg.differential) {
+			val_mv = (int32_t)((int16_t)buf);
+		} else {
+			val_mv = (int32_t)buf;
+		}
+		err = adc_raw_to_millivolts_dt(&adc_channels[i],&val_mv);
+		int val_int = (int)val_mv;
+		moisture = mapSoilMoisture(val_int, DryValue, WetValue, 0, 100);
+
+	}
+	return moisture;
+}
+```
+This function returns values of moisture in range 0-100.
+```c
+void set_led(uint8_t *data_recive, const struct device *gpio1, const struct device *gpio2){
+	int ret;
+	int SetLed = atoi(data_recive);
+	if (SetLed == 1) {
+		
+		ret = gpio_pin_set_raw(gpio2, 10, 0);
+		ret = gpio_pin_set_raw(gpio1, 0, 1);	
+			
+	} else {
+		
+		ret = gpio_pin_configure(gpio1, 5,GPIO_OUTPUT_ACTIVE); //red
+		if (ret != 0) {
+			return;
+		}
+		
+		ret = gpio_pin_set_raw(gpio1, 0, 0);
+		ret = gpio_pin_set_raw(gpio2, 10, 1);
+			
+	} 
+}
+```
+This function sets color of leds depending on the values send from dev board.
+## Finishing
+If you flashed your board with built programms you can check their logs.
+To see logs
+1. Open terminal
+2. Use command `sudo picocom --baud 115200 /dev/"your_usb_port"`
+### On Wio E5 dev board kit logs should looks like this:
+![Wio E5 dev board logs](./docs/asset/logs_dev_board.png)
+* Recive data informs us that data has been recived
+* Recived data shows value of moisture
+* Sending data infroms us that data has been sended
+* Data send shows value that has been send: 1 means you need too water the plant and 0 means that your soil moisture is correct
+* Blinking led on board informs us that there is connection via uart.
+### On Wio E5 mini board logs should looks like this:
+![Wio E5 dev board logs](./docs/asset/logs_mini_board.png)
```

Comment:
This is why I don't like pictures of text in the README. The images are not proportional and the size of letters varies a lot. Use:
```sh
blabla 
goo
foo
```

Reply:
Add extra line here